lib/msal-common/src/utils/ProtocolUtils.ts

```diff
@@ -34,4 +34,8 @@ export class ProtocolUtils {
         }
         return "";
     }
+
+    static setMethodInUserRequestState(userRequestState: string, method: MsalMethod): string {
+        return !StringUtils.isEmpty(userRequestState) ? `${userRequestState}${Constants.RESOURCE_DELIM}${method}` : method;
```

Comment:
If you wanted common to be use case agnostic, you could have this method in common take in library state (which each library will compute with its needed business logic) and user state, and just return a string that concatenates them together.

Reply:
i see. that makes sense. more extendable too. will edit